PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx

```diff
@@ -136,15 +136,31 @@ struct createEMReducedMCEvent {
         if (mctrack.pt() < 1e-2 || abs(mctrack.y()) > 1.5 || abs(mctrack.vz()) > 250 || sqrt(pow(mctrack.vx(), 2) + pow(mctrack.vy(), 2)) > 260) {
           continue;
         }
-
         int pdg = mctrack.pdgCode();
+
+        // if(pdg == 111){
+        //   if (mctrack.has_mothers()) {
+        //     //auto mck0s = mcTracks.iteratorAt(mctrack.mothersIds()[0]);
+        //     auto mp = mctrack.template mothers_first_as<aod::McParticles>();
+        //     int pdg_mp = mp.pdgCode();
+        //     if(pdg_mp == 310){
+        //       LOGF(info,"pi0 from K0S : index = %d , mc track pdg = %d , producedByGenerator =  %d , isPhysicalPrimary = %d", mctrack.index(), mctrack.pdgCode(), mctrack.producedByGenerator(), mctrack.isPhysicalPrimary());
+        //     }
+        //     else if(pdg_mp == 223){
+        //       LOGF(info,"pi0 from omega : index = %d , mc track pdg = %d , producedByGenerator =  %d , isPhysicalPrimary = %d", mctrack.index(), mctrack.pdgCode(), mctrack.producedByGenerator(), mctrack.isPhysicalPrimary());
+        //     }
+
+        //   }
+        // }
+
+        // Note that pi0 from weak decay gives producedByGenerator() = false
```

Comment:
why add this code if it is commented out?

Reply:
it was only test. I will remove these lines in the next PR.

Reply:
Okay :+1: